proxy/proxy.go

```diff
@@ -372,9 +372,13 @@ func copyStream(to flusherWriter, from io.Reader) error {
 // creates an outgoing http request to be forwarded to the route endpoint
 // based on the augmented incoming request
 func mapRequest(r *http.Request, rt *routing.Route, host string, removeHopHeaders bool) (*http.Request, error) {
+
 	u := r.URL
-	u.Scheme = rt.Scheme
-	u.Host = rt.Host
+
+	if rt.BackendType != eskip.ForwardBackend {
+		u.Scheme = rt.Scheme
+		u.Host = rt.Host
+	}
```

Comment:
I think the Scheme is never set. In my test:

```
/bin/skipper -inline-routes '* -> setRequestHeader("X-Passed-Skipper", "true") -> <forward>'
[APP]INFO[0000] Expose metrics in codahale format
[APP]INFO[0000] route settings, reset, route: : * -> setRequestHeader("X-Passed-Skipper", "true") -> <forward>
[APP]INFO[0000] route settings received
[APP]INFO[0000] route settings applied
[APP]INFO[0000] support listener on :9911
[APP]INFO[0000] proxy listener on :9090
[APP]INFO[0000] certPathTLS or keyPathTLS not found, defaulting to HTTP

[APP]ERRO[0004] error while proxying, route  with backend , status code 500: dialing failed false: unsupported protocol scheme ""
::1 - - [18/Jan/2019:14:44:53 +0100] "GET / HTTP/1.1" 500 22 "-" "curl/7.49.0" 0 localexample.org - -
[APP]ERRO[0013] error while proxying, route  with backend , status code 500: dialing failed false: unsupported protocol scheme ""
::1 - - [18/Jan/2019:14:45:02 +0100] "GET / HTTP/1.1" 500 22 "-" "curl/7.49.0" 0 www.google.com - -

[APP]ERRO[0022] error while proxying, route  with backend , status code 500: dialing failed false: unsupported protocol scheme ""
::1 - - [18/Jan/2019:14:45:11 +0100] "GET / HTTP/1.1" 500 22 "-" "curl/7.49.0" 0 www.google.com - -
```

Client calls:
```
% curl localhost:9090 -H"X-Sandor: yes" -H"Host: www.google.com" -v
* Rebuilt URL to: localhost:9090/
*   Trying ::1...
* Connected to localhost (::1) port 9090 (#0)
> GET / HTTP/1.1
> Host: www.google.com
> User-Agent: curl/7.49.0
> Accept: */*
> X-Sandor: yes
>
< HTTP/1.1 500 Internal Server Error
< Content-Type: text/plain; charset=utf-8
< Server: Skipper
< X-Content-Type-Options: nosniff
< Date: Fri, 18 Jan 2019 13:45:11 GMT
< Content-Length: 22
<
Internal Server Error
* Connection #0 to host localhost left intact
[sszuecs@sandor-lab:~/go/src/github.com/zalando/skipper/skptesting]% curl localhost:9090 -H"X-Sandor: yes" -H"Host: http://www.google.com" -v
* Rebuilt URL to: localhost:9090/
*   Trying ::1...
* Connected to localhost (::1) port 9090 (#0)
> GET / HTTP/1.1
> Host: http://www.google.com
> User-Agent: curl/7.49.0
> Accept: */*
> X-Sandor: yes
>
< HTTP/1.1 400 Bad Request: malformed Host header
< Content-Type: text/plain; charset=utf-8
< Connection: close
<
* Closing connection 0
400 Bad Request: malformed Host header
```

Reply:
Ok did it wrong as offline discussed and tested it works with `--proxy`:

backend:
```
% sudo ./bin/skipper -address :80 -inline-routes 'r: * -> inlineContent("OK") -> <shunt>'
[APP]INFO[0000] Expose metrics in codahale format
[APP]INFO[0000] support listener on :9911
[APP]INFO[0000] proxy listener on :80
[APP]INFO[0000] certPathTLS or keyPathTLS not found, defaulting to HTTP
[APP]INFO[0000] route settings, reset, route: r: * -> inlineContent("OK") -> <shunt>
[APP]INFO[0000] route settings received
[APP]INFO[0000] route settings applied
127.0.0.1 - - [18/Jan/2019:15:04:41 +0100] "GET / HTTP/1.1" 200 2 "-" "curl/7.49.0" 0 localexample.org - -

```

proxy:
```
% ./bin/skipper -inline-routes '* -> setRequestHeader("X-Passed-Skipper", "true") -> <forward>'
[APP]INFO[0000] Expose metrics in codahale format
[APP]INFO[0000] support listener on :9911
[APP]INFO[0000] proxy listener on :9090
[APP]INFO[0000] certPathTLS or keyPathTLS not found, defaulting to HTTP
[APP]INFO[0000] route settings, reset, route: : * -> setRequestHeader("X-Passed-Skipper", "true") -> <forward>
[APP]INFO[0000] route settings received
[APP]INFO[0000] route settings applied
::1 - - [18/Jan/2019:15:05:35 +0100] "GET http://localexample.org/ HTTP/1.1" 200 2 "-" "curl/7.49.0" 0 localexample.org - -
```

client
```
% curl --proxy localhost:9090 http://localexample.org -v
* Rebuilt URL to: http://localexample.org/
*   Trying ::1...
* Connected to localhost (::1) port 9090 (#0)
> GET http://localexample.org/ HTTP/1.1
> Host: localexample.org
> User-Agent: curl/7.49.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Length: 2
< Content-Type: text/plain; charset=utf-8
< Date: Fri, 18 Jan 2019 14:05:35 GMT
< Server: Skipper
<
* Connection #0 to host localhost left intact
OK
```